x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ShrinkAction.java

```diff
@@ -278,8 +300,15 @@ public List<Step> toSteps(Client client, String phase, Step.StepKey nextStepKey)
             ShrinkIndexNameSupplier::getShrinkIndexName
         );
         DeleteStep deleteSourceIndexStep = new DeleteStep(deleteIndexKey, isShrunkIndexKey, client);
-        ShrunkenIndexCheckStep waitOnShrinkTakeover = new ShrunkenIndexCheckStep(isShrunkIndexKey, nextStepKey);
-        return Arrays.asList(
+        ShrunkenIndexCheckStep waitOnShrinkTakeover = new ShrunkenIndexCheckStep(isShrunkIndexKey, lastOrNextStep);
+        UpdateSettingsStep allowWriteAfterShrinkStep = new UpdateSettingsStep(
+            allowWriteKey,
+            nextStepKey,
+            client,
+            CLEAR_WRITE_BLOCK_SETTINGS
+        );
```

Comment:
Minor, here we are creating the step no matter if we are going to use it or not. Consider only creating it if the flag requires it, by doing this you can also remove the flag check later. (Btw, I know it can be hard to spot things like this during incremental changes).

```
UpdateSettingsStep allowWriteAfterShrinkStep = allowWriteAfterShrink ? new UpdateSettingsStep(
            allowWriteKey,
            nextStepKey,
            client,
            CLEAR_WRITE_BLOCK_SETTINGS
        ) : null;
```